### PR TITLE
added periodCount in Group, changed userMapping, added new tests

### PR DESCRIPTION
**Pull request by khanhlinhtpnt**

### CODE REVIEW

File: contracts/TandaPay.sol [hunk 1]
```diff
@@ -54,75 +51,101 @@ contract Tandapay {
...
-    function startActivePeriod(uint groupId) public secretaryOnly(groupId) validId(groupId){
-        Group storage currentGroup = groups[groupId];
-                
-        require(currentGroup.paidPremiumCount == currentGroup.userList.length); //All premiums have been paid
-        require(currentGroup.newPeriod.phase == Phase.PRE); //Group is not active
-        
-        currentGroup.newPeriod.phase = Phase.ACTIVE;
-        currentGroup.currentPeriod = 2;
-        // Reset userMapping
-        for (uint i = 0; i < currentGroup.userList.length; i++) {
```
Comment (roslynlu) [inline contracts/TandaPay.sol:114]: Why did you delete this part that resets the user mapping? It seems necessary to me

Comment (khanhlinhtpnt) [reply contracts/TandaPay.sol:114]: I added periodCount so every time a user makes a payment, usermapping increments. so the mapping contains all users and their latest paid period as well. 
So we don't need to keep a separate array for users
